dynamiqs/utils/vectorization.py

```diff
@@ -115,7 +116,8 @@ def spre(x: QArrayLike) -> QArray:
     x = asqarray(x)
     check_shape(x, 'x', '(..., n, n)')
     n = x.shape[-1]
-    return eye(n) & x
+    xpre = eye(n) & x
```

Comment:
Not really useful but could have been written as:
```suggestion
    xpre = eye(*x.dims) & x
```
I guess we should mostly introduce a fast `__and__` for $I \otimes X$ and $X \otimes I$ as it's a bit dumb to initialize an identity matrix here.